src/main/java/net/minestom/server/entity/Entity.java

```diff
@@ -1718,8 +1722,29 @@ public boolean intersectBoxSwept(@NotNull Point rayStart, @NotNull Point rayDire
         return boundingBox.relativeEnd();
     }
 
-    public boolean hasCollision() {
-        return hasCollision;
+    public boolean isCollidesWithEntities() {
+        return collidesWithEntities;
+    }
+
+    public boolean isPreventBlockPlacement() {
+        // EntityMeta can change at any time, so initializing this during #initCollisions is not an option
+        // Can be overridden to allow for custom behaviour
+        if (entityMeta instanceof ArmorStandMeta armorStandMeta && armorStandMeta.isMarker()) return false;
+        return preventBlockPlacement;
+    }
+
+    protected void initCollisions() {
+        preventBlockPlacement = entityType != EntityType.ARROW
```

Comment:
You should probably make constants `Set<EntityType>` instead of chained expressions